src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int INITIAL_CAPACITY = 10;
+    private int capacity;
+    private int size;
+    private Object[] elements;
+
+    ArrayList() {
+        elements = new Object[INITIAL_CAPACITY];
+        capacity = INITIAL_CAPACITY;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size() == capacity) {
+            resize();
+        }
+        if (isEmpty()) {
+            elements[0] = value;
+            size++;
+        } else {
+            elements[size] = value;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("index is out of bounds of the list size");
+        }
+        if (size == capacity) {
+            resize();
+        }
+        size++;
+        for (int i = size - 1; i > index; i--) {
+            elements[i] = elements[i - 1];
+        }
```

Comment:
check common mistakes https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#use-systemarraycopy-to-move-your-array-elements